### ISSUE TITLE
ONNX: export failure: Exporting the operator silu to ONNX opset version 12 is not supported. 

### ISSUE THREAD

**Issue by gsx1378** [ultralytics/yolov5#7832]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

when i run "python export.py --weights ./models/download/yolov5n.pt --include onnx", the result is as follows:
`export: data=data/coco128.yaml, weights=['./models/download/yolov5n.pt'], imgsz=[640, 640], batch_size=1, device=cpu, half=False, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['onnx']
YOLOv5 🚀 v6.1-190-g4d59f65 Python-3.8.8 torch-1.7.0 CPU

Fusing layers... 
YOLOv5n summary: 213 layers, 1867405 parameters, 0 gradients, 4.5 GFLOPs

PyTorch: starting from models/download/yolov5n.pt with output shape (1, 25200, 85) (3.9 MB)

ONNX: starting export with onnx 1.9.0...
ONNX: export failure: Exporting the operator silu to ONNX opset version 12 is not supported. Please open a bug to request ONNX export support for the missing operator.`
How to solve this problem? Looking forward to your reply. ^_^

**Comment by Im-JimmyHu** [comment]
it looks like the the version of torch cause the error,may you can try to export with torch>=1.9,<=1.10.1,also compitable with the version onnx,wich i have successfullly convert the model to onnx.
if not solved,may you also try to replace the silu with this:
```
class SiLU(Module):
 
    __constants__ = ['inplace']
    inplace: bool
 
    def __init__(self, inplace: bool = False):
        super(SiLU, self).__init__()
        self.inplace = inplace
 
    def forward(self, input: Tensor) -> Tensor:
    
        return input * torch.sigmoid(input)
 

```

**Comment by glenn-jocher** [comment]
@gsx1378 it appears you may have environment problems. Your ONNX and PyTorch versions are both a bit old.

Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual [Python 3.9](https://www.python.org/) environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install` [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) again from scratch.

💡 ProTip! Try one of our verified environments below if you are having trouble with your local environment.


## Requirements

[**Python>=3.7.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

[Models](https://github.com/ultralytics/yolov5/tree/master/models) and [datasets](https://github.com/ultralytics/yolov5/tree/master/data) download automatically from the latest YOLOv5 [release](https://github.com/ultralytics/yolov5/releases) when first requested.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by ZCzzzzzz** [comment]
> it looks like the the version of torch cause the error,may you can try to export with torch>=1.9,<=1.10.1,also compitable with the version onnx,wich i have successfullly convert the model to onnx. if not solved,may you also try to replace the silu with this:
> 
> ```
> class SiLU(Module):
>  
>     __constants__ = ['inplace']
>     inplace: bool
>  
>     def __init__(self, inplace: bool = False):
>         super(SiLU, self).__init__()
>         self.inplace = inplace
>  
>     def forward(self, input: Tensor) -> Tensor:
>     
>         return input * torch.sigmoid(input)
>  
> ```

Can you tell me where this code should be placed?

**Comment by gsx1378** [comment]
> > it looks like the the version of torch cause the error,may you can try to export with torch>=1.9,<=1.10.1,also compitable with the version onnx,wich i have successfullly convert the model to onnx. if not solved,may you also try to replace the silu with this:
> > ```
> > class SiLU(Module):
> >  
> >     __constants__ = ['inplace']
> >     inplace: bool
> >  
> >     def __init__(self, inplace: bool = False):
> >         super(SiLU, self).__init__()
> >         self.inplace = inplace
> >  
> >     def forward(self, input: Tensor) -> Tensor:
> >     
> >         return input * torch.sigmoid(input)
> >  
> > ```
> 
> Can you tell me where this code should be placed?

"./utils/activations.py", You can search for it with this command: `grep -rni "SiLU(" ./ --binary-files=without-match`

**Comment by glenn-jocher** [comment]
@gsx1378 thank you for your question! Regarding the suggested code change, you can find the `Silu` function located in the `./models/common.py` file. As for the additional code provided, this `SiLU` implementation can be found in `./utils/activations.py`. You could find it by running this command: `grep -rni "SiLU(" ./ --binary-files=without-match` from the command line.

Please give this a try, and feel free to reach out if you need further assistance.
